tests/sentry/incidents/endpoints/test_project_alert_rule_details.py

```diff
@@ -237,6 +291,87 @@ def test_kicks_off_slack_async_job(
+            body={"ok": "true", "channel": {"name": channelName}},
+        )
+        resp = self._project_alert_rule_api_call(channelName, channelID)
+
+        stored_action = resp.data["triggers"][0]["actions"][0]
+        assert stored_action["inputChannelId"] == str(channelID)
+        assert stored_action["targetIdentifier"] == channelName
+
+    @responses.activate
+    def test_create_slack_alert_with_mistmatch_name_and_channel_id(self):
+        """
+        The user specifies the Slack channel and channel ID but they do not match.
+        """
+        otherChannel = "some-other-channel"
+        channelName = "my-channel"
+        # Specifying an inputChannelID will cause the validate_channel_id logic to be triggered
+        channelID = 123
+        self._mock_slack_response(
+            url=f"https://slack.com/api/conversations.info?channel={channelID}",
+            body={"ok": "true", "channel": {"name": otherChannel}},
+        )
+        resp = self._project_alert_rule_api_call(channelName, channelID)
+
+        assert resp.status_code == 400
+        assert resp.data == {
+            "nonFieldErrors": [
+                ErrorDetail(
+                    string=f"Received channel name {otherChannel} does not match inputted channel name {channelName}.",
+                    code="invalid",
+                )
+            ]
+        }
+
+    # An incorrect channelID will raise an ApiError in the Slack client
+    @patch.object(SlackClient, "get", side_effect=ApiError(text="channel_not_found"))
+    @responses.activate
+    def test_create_slack_alert_with_non_existent_channel_id(self, mock_slack_client):
+        """
+        The user specifies a bad Slack channel ID.
+        """
+        channelName = "my-channel"
+        # Specifying an inputChannelID will cause the validate_channel_id logic to be triggered
+        channelID = 123
+        resp = self._project_alert_rule_api_call(channelName, channelID)
+
+        assert resp.status_code == 400
+        assert resp.data == {
+            "nonFieldErrors": [
+                ErrorDetail(string="Channel not found. Invalid ID provided.", code="invalid")
+            ]
+        }
+
+    @patch("sentry.tasks.integrations.slack.find_channel_id_for_alert_rule.apply_async")
+    @patch("sentry.integrations.slack.utils.rule_status.uuid4")
+    @responses.activate
+    def test_create_slack_alert_with_empty_channel_id(
+        self, mock_uuid4, mock_find_channel_id_for_alert_rule
+    ):
+        """
+        The user selects the channel ID field and the UI will send the empty string to the
+        endpoint, thus, a channel name search will be performed
+        """
+        mock_uuid4.return_value = self.get_mock_uuid()
+        channelName = "my-channel"
+        # Because channel ID is an empty string it will be converted to an async request for the channel name
+        resp = self._project_alert_rule_api_call(channelName, channelID="")
```

Comment:
In this test we don't need to mock the object or add a response because it won't be needed to hit the Slack API as part of the API call (it will be queued).